safe-global/safe-wallet-web
# Feat(Multichain): add feature flag checks when adding a new network [SW-240]

## What it solves

Resolves: https://www.notion.so/safe-global/Add-feature-toggle-for-adding-networks-10e8180fe57380b4b27df647dae49742

## How this PR fixes it
- adds new feature flag `MULTI_CHAIN_SAFE_ADD_NETWORK`.
- Checks that the current safe network has the feature enabled anywhere we show the option to add a network.
- Checks that other networks have the feature enabled before showing them as an option to add to the group.


## How to test it
- When the feature flag is not enabled on the current chain, you should not see the option to add a network:
    - in the safe list context menu
    - at the bottom of multichain groups in the safe list
    - in the network selector
- When the feature flag is enabled on the current chain but not enabled on another compatible chain, disabled chains should not be shown:
    - in the network selector
    - in the add network modal

## Screenshots
![image](https://github.com/user-attachments/assets/d2d95cc2-c469-4b73-98c4-94410adc5769)


## Checklist
